packages/common/src/utils/SafeLocalStorage.ts

```diff
@@ -24,23 +34,29 @@ export const SafeLocalStorage = {
     value: SafeLocalStorageItems[Key]
   ): void {
     if (isSafe()) {
-      localStorage.setItem(key, JSON.stringify(value))
+      if (typeof value === 'string') {
+        localStorage.setItem(key, value)
+      } else {
+        localStorage.setItem(key, JSON.stringify(value))
+      }
     }
   },
-  getItem<Key extends keyof SafeLocalStorageItems>(key: Key): SafeLocalStorageItems[Key] | null {
+  getItem<Key extends keyof SafeLocalStorageItems>(
+    key: Key
+  ): SafeLocalStorageItems[Key] | undefined {
     if (isSafe()) {
       const value = localStorage.getItem(key)
 
       if (value) {
         try {
           return JSON.parse(value)
         } catch {
-          return value
+          return undefined
```

Comment:
I say we add a warning log or similar here, so that we know there was an error loading from localstorage